test/test_transforms_tensor.py

```diff
@@ -710,6 +569,134 @@ def test_autoaugment(self):
+    {"size": (5,)},
+    {"size": [5, ]},
+    {"size": (4, 5)},
+    {"size": [4, 5]}
+])
+def test_x_crop(func, method, out_length, fn_kwargs, device):
+    meth_kwargs = fn_kwargs
+
+    fn = getattr(F, func)
+    scripted_fn = torch.jit.script(fn)
+
+    tensor, pil_img = _create_data(height=20, width=20, device=device)
+    transformed_t_list = fn(tensor, **fn_kwargs)
+    transformed_p_list = fn(pil_img, **fn_kwargs)
+    assert len(transformed_t_list) == len(transformed_p_list)
+    assert len(transformed_t_list) == out_length
+    for transformed_tensor, transformed_pil_img in zip(transformed_t_list, transformed_p_list):
+        _assert_equal_tensor_to_pil(transformed_tensor, transformed_pil_img)
+
+    transformed_t_list_script = scripted_fn(tensor.detach().clone(), **fn_kwargs)
+    assert len(transformed_t_list) == len(transformed_t_list_script)
+    assert len(transformed_t_list_script) == out_length
+    for transformed_tensor, transformed_tensor_script in zip(transformed_t_list, transformed_t_list_script):
+        assert_equal(transformed_tensor, transformed_tensor_script)
+
+    # test for class interface
+    fn = getattr(T, method)(**meth_kwargs)
+    scripted_fn = torch.jit.script(fn)
+    output = scripted_fn(tensor)
+    assert len(output) == len(transformed_t_list_script)
+
+    # test on batch of tensors
+    batch_tensors = _create_data_batch(height=23, width=34, channels=3, num_samples=4, device=device)
+    torch.manual_seed(12)
+    transformed_batch_list = fn(batch_tensors)
+
+    for i in range(len(batch_tensors)):
+        img_tensor = batch_tensors[i, ...]
+        torch.manual_seed(12)
+        transformed_img_list = fn(img_tensor)
+        for transformed_img, transformed_batch in zip(transformed_img_list, transformed_batch_list):
+            assert_equal(transformed_img, transformed_batch[i, ...])
+
+
+@cpu_only
+@pytest.mark.parametrize('method', ["FiveCrop", "TenCrop"])
+def test_x_crop_save(method):
+    fn = getattr(T, method)(size=[5, ])
+    scripted_fn = torch.jit.script(fn)
+    with get_tmp_dir() as tmp_dir:
+        scripted_fn.save(os.path.join(tmp_dir, "t_op_list_{}.pt".format(method)))
+
+
+@pytest.mark.parametrize('device', cpu_and_gpu())
+@pytest.mark.parametrize('dt', [None, torch.float32, torch.float64])
+@pytest.mark.parametrize('size', [32, 34, [32, ], [32, 32], (32, 32), [34, 35]])
+@pytest.mark.parametrize('max_size', [None, 35, 1000])
+@pytest.mark.parametrize('interpolation', [BILINEAR, BICUBIC, NEAREST])
+def test_resize(dt, size, max_size, interpolation, device):
+
+    # TODO: Minimal check for bug-fix, improve this later
+    x = torch.rand(3, 32, 46)
+    t = T.Resize(size=38)
+    y = t(x)
+    # If size is an int, smaller edge of the image will be matched to this number.
+    # i.e, if height > width, then image will be rescaled to (size * height / width, size).
+    assert isinstance(y, torch.Tensor)
+    assert y.shape[1] == 38
+    assert y.shape[2] == int(38 * 46 / 32)
```

Comment:
This part should be extracted out as it doesn't benefit from the parametrization. As it comes from https://github.com/pytorch/vision/pull/2869/files, perhaps @vfdev-5 can chip in how we can improve this (as per the TODO)

Reply:
> By extracting out you mean to move them in a different test? right!?
I'm thinking of splitting them into two tests : test_resize_int and test_resize_scripted.

Yes! And yes, the naming sounds good to me!